apps/files/models.py

```diff
@@ -131,10 +131,15 @@ def get_mirror(self, addon, attachment=False):
         return posixpath.join(*map(smart_str, [host, addon.id, self.filename]))
 
     def get_url_path(self, src):
+        return self._make_download_url('downloads.file', src)
+
+    def get_signed_url_path(self, src):
+        return self._make_download_url('signing.file', src)
+
+    def _make_download_url(self, view_name, src):
         from amo.helpers import urlparams, absolutify
-        url = os.path.join(reverse('downloads.file', args=[self.id]),
+        url = os.path.join(reverse(view_name, args=[self.pk]),
```

Comment:
This shouldn't be using `os.path.join` for a URL should it? Wouldn't that make things backslashes if we started hosting this on Windows?

You could use `urlparse.urljoin` but the URL helpers always worry me because trailing slashes become meaningful.

It's probably fine, we won't ever host this on not-Unix but it's weird.